### DISCUSSION TITLE
UI Integration with keycloak identity provider

### DISCUSSION THREAD

**Discussion by egevorkyan** [longhorn/longhorn#3638] [Q&A]
Hello Team,

I want to ask question about integration longhorn UI with Keycloak Identity provider to secure Longhorn UI access. I have keykloak as entire SSO system for my Kubernetes, and really want to integrate it with Longhorn UI.

Any suggestions and recommendations are welcomed!!!

Thank you in advance

**Comment by innobead** [comment]
Suggest just leverage oauth2 proxy + ingress controller like Nginx ingress to fulfill your goals to authenticate longhorn UI access. Go ahead to try and share with us.

ref: 
- https://kubernetes.github.io/ingress-nginx/examples/auth/oauth-external-auth/
- https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider/#keycloak-oidc-auth-provider
